kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/V3ResourcesModule.java

```diff
@@ -30,5 +37,40 @@ protected void configure() {
     bindAsContract(ChunkedOutputFactory.class);
     bindAsContract(StreamingResponseFactory.class);
     bind(Clock.systemUTC()).to(Clock.class);
+    bindFactory(ExecutorServiceFactory.class, Singleton.class)
+        .to(ExecutorService.class)
+        .in(Singleton.class);
+  }
+
+  private static final class ExecutorServiceFactory implements Factory<ExecutorService> {
```

Comment:
nit: It probably makes sense to use a new marker interface which is a subtype of `ExecutorService`. This is not really an `ExecutorSeervice` provider, it's a `ProduceResponseExecutor` provider.

Reply:
+1, not for marker interface, but for annotation.

Reply:
I'm not sure I agree that it's not an executorService provider - all it does is create and return an executor service with the right config.

@rigelbm can you explain what you mean by annotation.  Looking at the other factories that do the same type of thing in the codebase, they are written very similarly to what is here so I don't understand the request.

Reply:
> I'm not sure I agree that it's not an executorService provider - all it does is create and return an executor service with the right config.

I think my main point is that we don't want to use this `ExecutorService` for generic purposes. It's used specifically for the blocking produce responses - e.g. it won't be a good idea to reuse the same instance for the AdminClient purposes that were mentioned above, but even if it wasn't the same instance, that "right" config will be right specifically for our case, not for other cases where we might need `ExecutorService`.

Reply:
@ehumber , by annotation, I mean what we do in `ConfigModule`, so that we inject it as `@MyAnnotation MyType` instead of just `MyType`.